tokio-timer/src/delay_queue.rs

```diff
@@ -690,24 +693,31 @@ impl<T> DelayQueue<T> {
     }
 }
 
-impl<T> Stream for DelayQueue<T> {
-    type Item = Expired<T>;
-    type Error = Error;
-
-    fn poll(&mut self) -> Poll<Option<Self::Item>, Error> {
-        let item = try_ready!(self.poll_idx()).map(|idx| {
-            let data = self.slab.remove(idx);
-            debug_assert!(data.next.is_none());
-            debug_assert!(data.prev.is_none());
-
-            Expired {
-                key: Key::new(idx),
-                data: data.inner,
-                deadline: self.start + Duration::from_millis(data.when),
-            }
-        });
+impl<T> Unpin for DelayQueue<T> {}
 
-        Ok(item.into())
+impl<T> Stream for DelayQueue<T> {
+    // DelayQueue seems much more specific, where a user may care that it
+    // has reached capacity, so return those errors instead of panicking.
+    type Item = Result<Expired<T>, Error>;
```

Comment:
Is my guess correct here? I've never used `DelayQueue`.

Reply:
that seems fine